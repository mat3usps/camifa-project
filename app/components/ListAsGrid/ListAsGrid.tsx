import type { ReactNode } from "react";

interface IListAsGridProps {
  children: ReactNode;
}

const ListAsGrid = ({ children }: IListAsGridProps) => {
  return (
    <div
      className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-4 lg:grid-cols-6"
      data-testid="ListAsGrid"
    >
      {children}
    </div>
  );
};

export default ListAsGrid;
