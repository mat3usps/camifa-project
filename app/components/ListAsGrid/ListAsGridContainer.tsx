import type { ReactNode } from "react";
import LinkButton from "~/components/button/LinkButton";
import InactiveItemsList from "~/components/InactiveItemsList/InactiveItemsList";
import ListAsGrid from "~/components/ListAsGrid/ListAsGrid";
import type { LinkTo } from "~/types/LinkTo";

export interface IListAsGridContainerProps<T> {
  activeList: T[];
  inactiveList: T[];
  inactiveTitle: string;
  registerFirstItem: {
    buttonLabel: string;
    title: string;
    to: LinkTo;
  };
  renderItem: (item: T) => ReactNode;
}

const ListAsGridContainer = <T,>({
  activeList,
  inactiveList,
  inactiveTitle,
  registerFirstItem,
  renderItem,
}: IListAsGridContainerProps<T>) => {
  return (
    <>
      {activeList.length < 1 && (
        <>
          <p className="font-bold">{registerFirstItem.title}</p>
          <LinkButton to={registerFirstItem.to} size="lg">
            {registerFirstItem.buttonLabel}
          </LinkButton>
        </>
      )}
      <ListAsGrid>{activeList.map(renderItem)}</ListAsGrid>
      <InactiveItemsList count={inactiveList.length} title={inactiveTitle}>
        {inactiveList.map(renderItem)}
      </InactiveItemsList>
    </>
  );
};

export default ListAsGridContainer;
