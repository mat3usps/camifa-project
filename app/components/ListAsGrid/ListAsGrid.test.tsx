import { render, screen } from "@testing-library/react";
import type { ComponentProps } from "react";
import ListAsGrid from "./ListAsGrid";

const { getByText } = screen;

describe("<ListAsGrid />", () => {
  it("should match snapshot", () => {
    const { container } = getRenderer({ children: <p>A children</p> });
    expect(container).toMatchSnapshot();
  });

  it.each(["A children", "Another children"])(
    "should render the children %p",
    (expected) => {
      getRenderer({
        children: <p>{expected}</p>,
      });
      expect(getByText(expected)).toBeInTheDocument();
    }
  );
});

// Helpers
function getRenderer({ children }: ComponentProps<typeof ListAsGrid>) {
  return render(<ListAsGrid>{children}</ListAsGrid>);
}
