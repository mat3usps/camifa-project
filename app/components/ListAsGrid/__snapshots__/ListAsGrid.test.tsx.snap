// Vitest Snapshot v1

exports[`<ListAsGrid /> > should match snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-4 lg:grid-cols-6"
    data-testid="ListAsGrid"
  >
    <p>
      A children
    </p>
  </div>
</div>
`;
